core/worker/src/main/java/com/cognifide/aet/worker/drivers/WebDriverProvider.java

```diff
@@ -23,42 +23,46 @@
 import com.google.common.base.Joiner;
 import com.google.common.collect.Maps;
 import java.util.Map;
-import org.apache.felix.scr.annotations.Activate;
-import org.apache.felix.scr.annotations.Component;
-import org.apache.felix.scr.annotations.Properties;
-import org.apache.felix.scr.annotations.Property;
-import org.apache.felix.scr.annotations.Reference;
-import org.apache.felix.scr.annotations.ReferenceCardinality;
-import org.apache.felix.scr.annotations.ReferencePolicy;
-import org.apache.felix.scr.annotations.Service;
+
 import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.metatype.annotations.Designate;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * @author lukasz.wieczorek
  */
-@Service(WebDriverProvider.class)
-@Component(label = "AET WebDriver Provider", description = "AET WebDriver Provider", immediate = true, metatype = true)
-@Properties({@Property(name = Constants.SERVICE_VENDOR, value = "Cognifide Ltd")})
+@Component(
+    service = WebDriverProvider.class,
+    property = {"name = " + Constants.SERVICE_VENDOR, "value = Cognifide Ltd"},
+    immediate = true
+)
+@Designate(ocd = WebDriverProviderConfig.class)
 public class WebDriverProvider {
 
   private static final Logger LOG = LoggerFactory.getLogger(WebDriverProvider.class);
 
-  private static final String DEFAULT_WEB_DRIVER_NAME = "defaultWebDriverName";
-
-  @Property(name = DEFAULT_WEB_DRIVER_NAME, label = "Default Web Driver name", value = "ff")
-  private String defaultWebDriverName;
+  private WebDriverProviderConfig webDriverProviderConfig;
```

Comment:
In PRs for other modules I've seen convention of naming those properties `config`. Could we please keep it also here and other classes in this PR?

Reply:
You mean changing this to 
`private WebDriverProviderConfig config`?

Reply:
Yes, and all other `XYZConfig` fields in other services.